### PR TITLE
Post Epidata v4 blog post

### PR DESCRIPTION
**Pull request by carlynvandyke**

### CODE REVIEW

File: content/blog/2022-12-01-epidata-v4.Rmd [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+We also added a "latest" table that includes a copy of just the issues that are most recent for all of our data points and removed the "latest issue" flag.  Since this no longer relies on looking in our gigantic collection of all issues for data that is flagged as being the latest, we gained a significant improvement to query times.  This is the data that drives our main [website](https://delphi.cmu.edu/covidcast/), so it is important for these operations to be fast for those browsing, and since we have so many visitors, it adds up to reduce our load significantly.  The "latest table" simplifies our data management responsibilities compared to the "latest flag", in that we don't have to clear a flag from an old row (a previously "latest" issue of data) when a new "latest" row is added -- instead we simply overwrite the old row in our latest table, and the database does this with great ease.
```
Comment (krivard) [inline content/blog/2022-12-01-epidata-v4.Rmd:71]: > We also added a "latest" table that includes a copy of just the issues that are most recent for all of our data points and removed the **"latest issue"** flag.

Should we use the same language here as in the _2. Data Ingestion/Repair and Development_ section? ("is_latest flag")

File: content/blog/2022-12-01-epidata-v4.Rmd [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+Querying historical data, e.g. for as-of, issue, or lag queries | 120% slower |
+
+Query time (seconds) | Latest | Historical |
+--- | --- | --- |
+Percentile | v3 | v4 | v3 | v4 |
+0th (minimum) | 0.005 | 0.005 | 0.005 | 0 .005 |
+25th | 0.007 | 0.006 | 0.006 | 0.007 |
+50th (median) | 0.009 | 0.008 | 0.007 | 0.008 |
+75th | 0.058 | 0.017 | 0.012 | 0.023 |
+100th (maximum) | 739.332 | 20.069 | 17.121 | 15.997 |
```
Comment (krivard) [inline content/blog/2022-12-01-epidata-v4.Rmd:96]: @minhkhul these tables are not super legible in the rendered website:
image

Can you experiment with styling options to resolve this? Goals are:
* easily tell which text belongs to which row (use borders, background color, whatever you think is good)
* reduce unnecessary line-wrapping

File: content/blog/2022-12-01-epidata-v4.Rmd [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+For historical data, long-running queries were more frequent in v4 than in v3, but not more severe. In v3, only 8 queries required more than one second to compute, and the longest-running query took seventeen seconds; in v4 over two hundred queries needed more than one second, but the max query time was sixteen seconds. 
+
+The empirical cumulative distribution functions for query times in the latest and historical datasets shows both of these effects in greater detail.
+
+Long-running queries (counts) | Latest (of 37,396) | Historical (of 19,359) |
+--- | --- | --- |
+Queries slower than | v3 | v4 | v3 | v4 |
+1 second | 2734 | 17 | 8 | 228 |
+10 seconds | 706 | 3 | 1 | 1 |
+100 seconds | 3 | 0 | 0 | 0 |
```
Comment (krivard) [inline content/blog/2022-12-01-epidata-v4.Rmd:109]: @minhkhul same problem, plus we want a colspan of 2 on the Latest and Historical headers so all four columns of numbers appear:

image

**Review by krivard — COMMENTED** [review]
Some questions for both @carlynvandyke and @minhkhul 

Also I need to update my author blurb, it's out of date. I'll push up a commit for that.

File: content/people/index.md [hunk 4]
```diff
@@ -139,6 +139,8 @@ people:
   team:
   - core
   - highlight
+  - blog
+  description: specializes in Delphi's compute infrastructure, and is a DevOps Engineer in the Machine Learning Department at CMU.
```
Comment (krivard) [inline content/people/index.md:143]: @korlaxxalrok I wrote this off the cuff, please check it and adjust as you like

File: content/people/index.md [hunk 5]
```diff
@@ -262,6 +264,8 @@ people:
   affiliation: CMU/CSD
   team:
   - core
+  - blog
+  description: specializes in Delphi's database and API, and is a Systems Software Engineer in the Machine Learning Department at CMU.
```
Comment (krivard) [inline content/people/index.md:268]: @melange396 I wrote this off the cuff, please check it and adjust as you like

File: content/blog/2022-12-01-epidata-v4.Rmd (hunk 2 above)
Comment (minhkhul) [reply content/blog/2022-12-01-epidata-v4.Rmd]: Screenshot 2023-01-24 at 11 52 56 AM

File: content/blog/2022-12-01-epidata-v4.Rmd (hunk 3 above)
Comment (minhkhul) [reply content/blog/2022-12-01-epidata-v4.Rmd]: Screenshot 2023-01-24 at 11 53 40 AM

**Comment by melange396** [comment]
for reference, the gdocs draft is [here](https://docs.google.com/document/d/1TAXPzDTCUBSknArNpCZfUPmBTlWAN9qnKgWoBBNedJ4/edit)

**Comment by melange396** [comment]
is there a way to do fixed-width fonts on this?

File: content/people/index.md (hunk 5 above)
Comment (melange396) [inline content/people/index.md:268]:
```suggestion
  description: specializes in Delphi's database and API, and is a Software Engineer in the Machine Learning Department at CMU.
```

File: content/blog/2022-12-01-epidata-v4.Rmd (hunk 1 above)
Comment (melange396) [reply content/blog/2022-12-01-epidata-v4.Rmd:71]: `"latest issue"` also appears elsewhere in the document (line 61 above).  we could do s/"latest issue"/is_latest/ here, but i think it looks a little silly without standing out in a fixed-width font or similar.  anyone reading and actually paying attention should be able to recognize that this "latest issue" flag is the same as the is_latest flag.  TL;DR: probbly leave it as-is

File: content/blog/2022-12-01-epidata-v4.Rmd [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+* Slow latest queries:
+  + Some data sources experience heavy revisions. The design of the v3 system required the database to scan all available issues of each matching reference date before it could identify the latest issue. The more revisions it needed to discard, the more time needed for the scan.
+	
+* COVIDcast Metadata:
+  + We compile a list of all COVIDcast-related sources and signals available in the API, along with basic summary statistics such as the dates they are available, their minimum and maximum values, and the geographic levels at which they are reported. This is made available here.
```
[the remark this answers is not in the document]
Comment (melange396) [reply content/blog/2022-12-01-epidata-v4.Rmd:37]: in the gdoc, "here" links to https://api.covidcast.cmu.edu/epidata/covidcast/meta

[the remark this answers is not in the document]
Comment (minhkhul) [reply content/blog/2022-12-01-epidata-v4.Rmd:37]: fixed.
Screenshot 2023-01-25 at 9 32 29 AM

File: content/blog/2022-12-01-epidata-v4.Rmd (hunk 1 above)
Comment (minhkhul) [reply content/blog/2022-12-01-epidata-v4.Rmd]: Added fixed-width font, change a bit of wording to show both "latest issue" and ``is_latest``
Screenshot 2023-01-25 at 9 53 54 AM
Screenshot 2023-01-25 at 9 54 23 AM

File: content/people/index.md (hunk 4 above)
Comment (korlaxxalrok) [reply content/people/index.md:143]: Looks good!

**Comment by melange396** [comment]
i pushed changes for other fixed-width stuff that i thought was appropriate, but im not married to it if there are objections.  with or without that, this looks good to me (modulo regenerating the html, of course)

**Review by melange396 — APPROVED** [review]

**Comment by melange396** [comment]
ugh...  looking at the netlify preview linked above, it seems spaces inside of fixed-width text (or `<code>` tags in the html) are non-breaking, so all of each segment of f-w text will get displayed on a single line, which screws up our paragraph structure (and i think it looks stupid) : 

image

i previewed doing a split of the f-w around spaces (by doing a manual/live edit of the elements) and think it looks better this way:

image

in a few minutes, ill push an update to this branch to fix that.

**Comment by minhkhul** [comment]
Made commas and parenthesis not fixed width for clarity. 
Screenshot 2023-01-26 at 12 53 00 PM

**Comment by melange396** [comment]
i kinda liked having the parens in red too because it made the tuple look like a whole "unit", and with regular parens it looks like its an aside instead.  on the other hand, at this point, im fine with whatever...  ¯\\\_(ツ)_/¯

**Review by krivard — APPROVED** [review]
:+1:

**Pull request merged**